pulsar-broker/src/main/java/org/apache/pulsar/zookeeper/MaxValueMetadataNodePayloadLenEstimator.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.zookeeper;
+
+import io.netty.util.concurrent.FastThreadLocal;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pulsar.metadata.api.GetResult;
+import org.apache.pulsar.metadata.api.MetadataNodePayloadLenEstimator;
+
+@Slf4j
+public class MaxValueMetadataNodePayloadLenEstimator implements MetadataNodePayloadLenEstimator {
+
+    // Default to max int value, let the first query command do not execute with batch.
+    public static final int DEFAULT_LEN = Integer.MAX_VALUE;
+    public static final int UNSET = -1;
+    // xid: int, zxid: long, err: int, len: int -> 20
+    // Node stat:
+    //   czxid: long, mzxid: long, ctime: long, mtime: long, version: int, cversion: int, aversion: int, -> 44
+    //   ephemeralOwner: long, dataLength: int, numChildren: int, pzxid: long -> 24
+    // total: 88, let's double it to cover different serialization versions.
+    public static final int ZK_PACKET_SYSTEM_PROPS_LEN = 176;
+    private static final SplitPathRes MEANINGLESS_SPLIT_PATH_RES = new SplitPathRes();
+    private static final FastThreadLocal<SplitPathRes> LOCAL_SPLIT_PATH_RES = new FastThreadLocal<SplitPathRes>() {
+        @Override
+        protected SplitPathRes initialValue() {
+            return new SplitPathRes();
+        }
+    };
+    private final int[] maxLenOfGetMapping;
+    private final int[] maxLenOfListMapping;
+
+    {
+        int pathTypeCount = PathType.values().length;
+        maxLenOfGetMapping = new int[pathTypeCount];
+        maxLenOfListMapping = new int[pathTypeCount];
+        for (int i = 0; i < pathTypeCount; i++) {
+            maxLenOfGetMapping[i] = UNSET;
+            maxLenOfListMapping[i] = UNSET;
+        }
+    }
```

Comment:
could we use ordinary constructor syntax here?

Reply:
Could you show an example?